contracts/Tokens/VTokens/VToken.sol

```diff
@@ -708,7 +708,8 @@ contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
         /* exchangeRate = invoke Exchange Rate Stored() */
         (vars.mathErr, vars.exchangeRateMantissa) = exchangeRateStoredInternal();
         if (vars.mathErr != MathError.NO_ERROR) {
-            return failOpaque(Error.MATH_ERROR, FailureInfo.REDEEM_EXCHANGE_RATE_READ_FAILED, uint(vars.mathErr));
+            failOpaque(Error.MATH_ERROR, FailureInfo.REDEEM_EXCHANGE_RATE_READ_FAILED, uint(vars.mathErr));
+            revert();
```

Comment:
We can't get events from failed txs AFAIK, so I propose just we just do this for now:

```suggestion
            revert("math error");
```

In the long run, we should upgrade the compiler and use custom errors IMO

Reply:
Still we would get events there because I did not remove `failOpaque(Error.MATH_ERROR, FailureInfo.REDEEM_EXCHANGE_RATE_READ_FAILED, uint(vars.mathErr));` @kkirka @coreyar

Reply:
My understanding is that events are not emitted if a transaction is reverted. There seem to be two tests for this - `returns error from redeemFresh without emitting any extra log`

Reply:
In this case we can remove the `failOpaque` calls when using revert